org/postgresql/core/Parser.java

```diff
@@ -362,4 +430,257 @@ private static boolean subArraysEqual(final char[] arr,
     
         return true;
     }
+
+    /**
+     * Contains parse flags from {@link #modifyJdbcCall(String, boolean, int, int)}.
+     * Originally {@link #modifyJdbcCall(String, boolean, int, int)} was located in {@link org.postgresql.jdbc2.AbstractJdbc2Statement},
+     * however it was moved out to avoid parse on each prepareCall.
+     */
+    public static class JdbcCallParseInfo {
+        private String sql;
+        private boolean isFunction;
+        private boolean outParmBeforeFunc;
+
+        public String getSql() {
+            return sql;
+        }
+
+        public boolean isFunction()
+        {
+            return isFunction;
+        }
+
+        public boolean isOutParmBeforeFunc()
+        {
+            return outParmBeforeFunc;
+        }
+    }
+
+    /**
+     * this method will turn a string of the form
+     * { [? =] call <some_function> [(?, [?,..])] }
+     * into the PostgreSQL format which is
+     * select <some_function> (?, [?, ...]) as result
+     * or select * from <some_function> (?, [?, ...]) as result (7.3)
+     */
+    public static JdbcCallParseInfo modifyJdbcCall(String p_sql, boolean stdStrings, int serverVersion, int protocolVersion) throws SQLException
```

Comment:
On 23 June 2015 at 17:10, Vladimir Sitnikov <notifications@github.com>
wrote:

> In org/postgresql/core/Parser.java
> <https://github.com/pgjdbc/pgjdbc/pull/319#discussion_r33092887>:
>
> > +        }
> > +
> > +        public boolean isOutParmBeforeFunc()
> > +        {
> > +            return outParmBeforeFunc;
> > +        }
> > +    }
> > +
> > +    /**
> > +     * this method will turn a string of the form
> > +     * { [? =] call <some_function> [(?, [?,..])] }
> > +     * into the PostgreSQL format which is
> > +     * select <some_function> (?, [?, ...]) as result
> > +     * or select * from <some_function> (?, [?, ...]) as result (7.3)
> > +     */
> > +    public static JdbcCallParseInfo modifyJdbcCall(String p_sql, boolean stdStrings, int serverVersion, int protocolVersion) throws SQLException
>
>  p_sql does not conform to Java naming conventions for parameters.
>
> This naming existed before. Do you think I should correct it within the
> current change?
>

Yes, please change it



Dave Cramer